_licenses/cecill-2.1.txt

```diff
@@ -0,0 +1,580 @@
+---
+title: CeCILL Free Software License Agreement v2.1
+spdx-id: CECILL-2.1
+source: http://www.cecill.info/licences/Licence_CeCILL_V2.1-fr.txt
+
+description: French open-source license made by the CEA, INRIA and CNRS, compatible with the GNU GPL v2.0 and v3.0
+
+how: Create a text file (typically named LICENSE or LICENCE) in the root of your source code and copy the text of the license into the file. 
+
+using:
+  - VITAM: https://github.com/ProgrammeVitam/vitam/blob/master_0.15.x/Licence_CeCILL_V2.1-en.txt
+  - Test-Compta-Demat: https://github.com/DGFiP/Test-Compta-Demat/blob/master/src/testeur/Licence_CeCILL_V2.1-fr.txt
+  - BMC-Tools: https://github.com/ANSSI-FR/bmc-tools/blob/master/LICENCE.txt
+
+permissions:
+  - commercial-use
+  - modifications
+  - distribution
+  - private-use
+  - patent-use
+
+conditions:
+  - include-copyright
+  - disclose-source
+  - same-license
+  
+limitations:
+  - liability
+  - warranty
+
+---
+
+
+  CONTRAT DE LICENCE DE LOGICIEL LIBRE CeCILL
+
+Version 2.1 du 2013-06-21
+
+
+    Avertissement
+
+Ce contrat est une licence de logiciel libre issue d'une concertation
+entre ses auteurs afin que le respect de deux grands principes préside à
+sa rédaction:
+
+  * d'une part, le respect des principes de diffusion des logiciels
+    libres: accès au code source, droits étendus conférés aux utilisateurs,
+  * d'autre part, la désignation d'un droit applicable, le droit
+    français, auquel elle est conforme, tant au regard du droit de la
+    responsabilité civile que du droit de la propriété intellectuelle et
+    de la protection qu'il offre aux auteurs et titulaires des droits
+    patrimoniaux sur un logiciel.
+
+Les auteurs de la licence CeCILL (Ce[a] C[nrs] I[nria] L[ogiciel] L[ibre])
```

Comment:
This obtains the following failure:

```
 1) license fillable fields The CeCILL Free Software License Agreement v2.1 license should only contain supported fillable fields
     Failure/Error: expect(extra_fields).to be_empty
       expected `["a", "nrs", "nria", "ogiciel", "ibre"].empty?` to return true, got false
     # ./spec/license_fields_spec.rb:11:in `block (4 levels) in <top (required)>'
```

I'm not sure what to do with this:

- change the license text, perhaps replace square brackets with angle brackets? a bit ugly
- remove test: does it really matter that there are things that look like fields with keys we don't know about? nothing will be done with them, right?

💭 @benbalter ?

Reply:
@benbalter can I escape the brackets? At the same time I don't want to change the license text or make sure it will be recognized. Thank you very much for your help.

Reply:
Ooph. What if we change the regex in 

https://github.com/github/choosealicense.com/blob/c229dccb6d356729eecfe4ac20cdf26906c1e29e/spec/license_fields_spec.rb#L9

to 

```ruby
/\s+\[([a-z]+)\]\s+/
```

to require the field be surrounded by whitespace?

The reason for the test was because some licenses had `name` and others had `author`, meaning in GitHub.com the inconsistency would cause the fields not to be interpolated.

Reply:
Hello, are you sure that for the others licenses' text, the fields would always be surrounded by space? For the CeCILL, that would do the trick, since the description has always a letter before.

Reply:
@benbalter I made your suggested change at d01065b09a3bbf9918f0b46f38e38a3d4aa7eb85 with you as the author. Please scream if you hate that.

@ljoubert I was, but OFL-1.1 has `([email])`. I'm not sure that _should_ be included -- it isn't in https://spdx.org/licenses/OFL-1.1.html while http://scripts.sil.org/cms/scripts/page.php?site_id=nrsi&id=OFL and https://opensource.org/licenses/OFL-1.1 include more headers/fields. Anyway that's a separate issue.

Also, various fields in various licenses are at the beginning or end of a line.

So I suppose the regex should be loosened a bit, and there should be a test to make sure there are no expected fields being excluded.

Reply:
Attempt at 286357a54c87b3adaf7ae5c5314275626f31b977